sunpy/visualization/animator/line.py

```diff
@@ -98,36 +98,32 @@ def __init__(self, data, plot_axis_index=-1, axis_ranges=None, ylabel=None, xlab
         if data.ndim < 2:
             raise ValueError("data must have at least two dimensions.  One for data "
                              "for each single plot and at least one for time/iteration.")
-        # Ensure axis_ranges are input correctly.
-        if axis_ranges is not None:
-            if axis_ranges[self.plot_axis_index] is not None:
-                if (len(axis_ranges[self.plot_axis_index]) != data.shape[self.plot_axis_index] and
-                        axis_ranges[self.plot_axis_index].shape != data.shape):
-                    raise ValueError("The plot_axis_index axis range must be specified as None "
-                                     "a 1D array of same length as plot_axis_index axis, "
-                                     "or an array of same shape as data.")
         # Define number of slider axes.
         self.naxis = data.ndim
         self.num_sliders = self.naxis-1
+
         # Attach data to class.
         if axis_ranges is not None and all(axis_range is None for axis_range in axis_ranges):
             axis_ranges = None
-        if axis_ranges is None or axis_ranges[self.plot_axis_index] is None:
-            self.xdata = np.arange(data.shape[self.plot_axis_index])
-        else:
-            self.xdata = np.asarray(axis_ranges[self.plot_axis_index])
         if ylim is None:
             ylim = (data.min(), data.max())
         if xlim is None:
-            xlim = (self.xdata.min(), self.xdata.max())
+            xlim = (axis_ranges[plot_axis_index].min(), axis_ranges[plot_axis_index].max())
         self.ylim = ylim
         self.xlim = xlim
         self.xlabel = xlabel
         self.ylabel = ylabel
+
         # Run init for base class
         super(LineAnimator, self).__init__(data, image_axes=[self.plot_axis_index],
                                            axis_ranges=axis_ranges, **kwargs)
 
+        # Attach data to class.
+        if axis_ranges is None or axis_ranges[self.plot_axis_index] is None:
```

Comment:
```suggestion
        if None in (axis_ranges, axis_ranges[self.plot_axis_index]):
```

Reply:
Though... looking it in more detail I'm not sure this will work... as it's looking into `axis_ranges` and then inside of `axis_ranges`. So my suggestion probably fails if `axis_ranges` is `None`.

Reply:
Which can be the case.